components/sidebar/sidebar.jsx

```diff
@@ -207,6 +222,8 @@ export default class Sidebar extends React.PureComponent {
         this.updateTitle();
         this.setBadgesActiveAndFavicon();
         this.setFirstAndLastUnreadChannels();
+
+        this.setChannelIds(snapshot, this.props, prevState);
```

Comment:
this looks dangerous as you are setting the state in componentDidUpdate

Reply:
I believe the danger is to not do it conditionally which I made sure to put it. This is the only lifecycle where I have access to snapshot and prevState.

Reply:
I'm not convinced, why would you want to re-render the component again at this point? I guess `setFirstAndLastUnreadChannels` is also doing it, and I'm going to approve but this looks wrong to me

Reply:
Okay, I think I'm getting what you mean, let me take a look again.

Reply:
Can we use `getDerivedStateFromProps` instead here so we don't trigger another render cycle? and this update gets called on of different prop values so if we can move this to `getDerivedStateFromProps` we might avoid this.

Reply:
You're right.  I'm updating this with `getDerivedStateFromProps`, will submit shortly.